### ISSUE TITLE
influxdb & grafana, absence of data on the graphs

### ISSUE THREAD

**Issue by oshamin** [influxdata/influxdb#8050]
__System info:__ 
OS: CentOS Linux release 7.3.1611 (Core)
influxdb: influxdb-1.2.0-1.x86_64 (https acccess)
grafana: grafana-4.1.2-1486989747.x86_64

__Steps to reproduce:__
I faced the problem, that grafana shows always "InfluxDB Error: undefined" after influxdb works for awhile.
Here is how it is presented, you can see exclamation mark on the upper left side with error (InfluxDB Error: undefined).

Dashboard loads
![refresh1](https://s23.postimg.org/7nbewddgb/image.png)

Waited ~2 seconds and clicked on refresh

![refresh2](https://s23.postimg.org/occusaa1n/image.png)

So after clicking refresh some graphs show data and some not.  
The InfluxQL queries are registered in logs without errors.
But in grafana logs some queries have response code 200 and some have 502, when queries are proxied via grafana.

__Additional info:__
If I only open one graph, then response is always successful.
Moreover the problem disappears after influxdb restarted and then appears after 5-10 minutes again.

my config is [here](https://gist.githubusercontent.com/oshamin/aef4dbf095ddce13c8b55b0006ee5dae/raw/09742f67160f49605074f6d09adba2d12ce843a2/gistfile1.txt)

What can be a cause of such inconsistent behavior?

**Comment by jwlighting** [comment]
I had the same problem discribed above with OS Ubuntu 16.10 after the update from InfluxB version 1.1 to 1.2. As a solution I downgraded to 1.1 again, which is working without problems

**Comment by jwilder** [comment]
A 502 might indicate that the proxy timed out during the query. Have you tried switch to querying influxdb directly?  Also, are there any errors in the influxdb logs associated with those queries?  What are the actual queries you are running that are returning 502 from grafana?

**Comment by oshamin** [comment]
@jwlighting my running influxdb instance was also updated from 1.1 to 1.2 version.

@jwilder yes, I have tested both direct and proxy access to influxdb. 
The difference is that with proxy access 502 errors are recorded in grafana logs and in browser console. 
![browser logs](https://s11.postimg.org/go2qg02mr/image.png)
And with direct access they are only in browser console.

**Comment by jwilder** [comment]
@oshamin Are there errors in the influxdb logs?  Can you include the logs for those queries?

**Comment by oshamin** [comment]
@jwilder It looks like not all queries come to influxdb. I made [video](https://youtu.be/BCfe_SrVMzk).
The dashboard had 6 graphs which were build by 9 queries, but influxdb only logged 1-3 query per refresh.

Then there are two possibilities, either queries are not logged properly or it is grafana bug.

How do you think should I create issue in grafana tracker?

**Comment by jwilder** [comment]
@oshamin Can you enable HTTP logging on influxdb?

```
[http]
  log-enabled = true
```

**Comment by oshamin** [comment]
here is [log](https://gist.githubusercontent.com/oshamin/4428e5222364992f6923c489cd35ab22/raw/e825cb985023fb788442b049ffdcf93e3904f7db/txt).

There are only one http request per refresh.

**Comment by jwilder** [comment]
@oshamin Try w/o the proxy config so that grafana is not in the middle of the browser and the databases.  If there is no HTTP log for the failed queries, then the query is not reaching influxdb for some reason.

**Comment by oshamin** [comment]
@jwilder the result is the same either via proxy or direct access. And not all http queries are logged.
But after influxdb restart everything works smoothly about 1-5 minutes and then the problem appears again. 

I also checked connections with tcpdump. After refreshing page, tcpdump registers more then one new TCP SYNC. And here is connection view from browser side.

![chrome connections](https://s10.postimg.org/no2oag3o9/image.png)

**Comment by graphex** [comment]
I'm seeing the same sorts of things happening with 1.2 and grafana. In one graph I can see up to 7 lines (each one has a different tag value), but no more. When I look at the raw json being returned, I see `partial: true` where it stops. Is this a new 'feature' that grafana doesn't know how to deal with? Is it configurable?

**Comment by oshamin** [comment]
After I gave up testing different version of influxdb.
I started looking closer to grafana. I disabled ssl and removed content of session folder /var/lib/grafana/sessions and cleared browser cache, then restarted grafana and the problem disappears.

Right now grafana works again with ssl and I do not have any errors either accessing influxdb via proxy or direct. 

As with influxdb, grafana was also periodically updated on the server
grafana-4.0.2-1481203731.x86_64 <- first install
grafana-4.1.1-1484211277.x86_64
grafana-4.1.2-1486989747.x86_64 <- now

As a markers for the problem, following errors were found in grafana/system log:
```log
Feb 28 22:12:42 ent grafana-server: 2017/02/28 22:12:42 http: proxy error: read tcp 127.0.0.1:35822->127.0.0.1:8086: read: connection reset by peer
Feb 28 22:12:42 ent grafana-server: 2017/02/28 22:12:42 http: proxy error: EOF
Feb 28 22:12:42 ent grafana-server: 2017/02/28 22:12:42 http: proxy error: read tcp 127.0.0.1:35836->127.0.0.1:8086: read: connection reset by peer
Feb 28 22:12:42 ent grafana-server: 2017/02/28 22:12:42 http: proxy error: read tcp 127.0.0.1:35834->127.0.0.1:8086: read: connection reset by peer
Feb 28 22:12:42 ent grafana-server: 2017/02/28 22:12:42 http: proxy error: read tcp 127.0.0.1:35832->127.0.0.1:8086: read: connection reset by peer
```

If the problem does not come again in the next few days, then I will suppose the problem with incorrect session.

**Comment by oshamin** [comment]
@jwilder connection in console also does not work.

```bash
$ influx -precision rfc3339
Failed to connect to http://localhost:8086: Get http://localhost:8086/ping: EOF
Please check your connection settings and ensure 'influxd' is running.

$ ps aux|grep influxdb
influxdb 30356  6.5  2.6 497440 157016 ?       Ssl  01:05   0:40 /usr/bin/influxd -config /etc/influxdb/influxdb.conf
```

and in log file only write request are logged (POST).
After restart it works for awhile and the problem repeat again.

How can I help find cause of the problem, what information do you need?

**Comment by gerrickw** [comment]
Same problem.  Before I stopped using 1.2.0 I noticed the same issue, but didn't research too much as had to revert due to '_' issue and 1.1.4 started working again.  Would like to move to 1.2.1 so here are some repro steps.

Steps to reproduce:
1. Install Grafana
2. Install influxdb 1.1.4
3. Setup graph in grafana with the following query:
SELECT max(value) FROM test_db.autogen./^f$/ WHERE pool =~ /^hhh_iii$/ AND colo =~ /^colob$/ AND time > now() - 1h GROUP BY time(10s), y, pool
4. Download python gist I made for another influxdb issue a while back.  https://gist.github.com/gerrickw/f83fb4d4d69aef2dfd37/6975818e9d6d3ef4ce248f57891ecb112f33edd0
5. pip install influxdb
6. Execute influx
7. > CREATE DATABASE "test_db"
8. > CREATE USER "test_db" WITH PASSWORD 'test_db'
9. > grant read on test_db to test_db
10. > grant write on test_db to test_db
11. python simple_influx_writer.py --hhh_iii -x 50 -y 80 -s 20000
11a. python simple_influx_writer.py --help for options.
12. Probably doesn't need to be that much data, but this is what I did to reproduce.
13. Run for an hour.
14. In graph in grafana notice for an hour period you have 80 Ys.
15. Stop script and upgrade influxdb to 1.2.1
16. Execute script again.
17. Look at graph in Grafana for an hour and notice you only see 20ish Ys.  Reduce time to 30 mins you see 40-50 Ys.  Reduce to 5 mins you will see all 80 Ys.

**Comment by oshamin** [comment]
Hi @gerrickw, what do you mean by "Ys."? Can you show image?

**Comment by gerrickw** [comment]
Grouping by Ys.  In this case from the script there are 50 Xs by 80 Ys.  So each Y is an aggregated view of max() of all of the Xs. So a large grid or shards that is showing the max value of the largest X of the Y.  Images showing the same query with only changing the amount of time queried in the graph. Notice the legend showing suddenly more data, even though the same query.  With influxdb 1.1.4, the legend always shows 80 Ys no matter the time range.

![1-hour](https://cloud.githubusercontent.com/assets/656611/23814343/0f10454c-0598-11e7-9145-217011b87b2a.png)
![30-minutes](https://cloud.githubusercontent.com/assets/656611/23814342/0f0fec00-0598-11e7-93e2-04f297a96d34.png)
![5-minutes](https://cloud.githubusercontent.com/assets/656611/23814341/0f0f0376-0598-11e7-929e-d83239c1cde0.png)

**Comment by jwilder** [comment]
@gerrickw Can you see if setting:

```
[http]
max-row-limit = 0
```

resolves that?

**Comment by gerrickw** [comment]
@jwilder Setting to 0 no longer causes the issue.  Hmm, seems odd to suddenly make a default.  This makes it appear it is a bug with influx not giving all of the data or possibly make the user believe there are no other values and drawing incorrect conclusions.  But this fixed my issue, so possibly this isn't related to main issue here.

**Comment by elvarb** [comment]
Here is the Grafana ticket on this issue, https://github.com/grafana/grafana/issues/7380

**Comment by elvarb** [comment]
It seems to be how InfluxDB now returns the json results in chunks, which is good but Grafana does not support it. 

Is there some way to disable the chunked feature?

**Comment by elvarb** [comment]
```
[http]
max-row-limit = 0
```
does indeed solve this

**Comment by jwilder** [comment]
Fixed via #8133
